jni/portolan_sync_api.cpp

```diff
@@ -72,18 +72,19 @@ Java_org_openobservatory_measurement_1kit_jni_sync_PortolanSyncApi_sendProbe
         if (ptr == 0L) throw std::runtime_error("Null pointer");
         ProberContext *ctx = (ProberContext *) ptr;
         std::string payload(payload_size, '\0');
-        ctx->prober.on_result([&ctx, &result](ProbeResult r) {
-            result = r;
-            ctx->poller.break_loop();
+        ctx->reactor->loop_with_initial_event([&]() {
+            ctx->prober.on_result([&ctx, &result](ProbeResult r) {
+                result = r;
+                ctx->reactor->break_loop();
+            });
+            ctx->prober.on_timeout([&ctx, &is_timed_out]() {
+                ctx->reactor->break_loop();
+                is_timed_out = true;
+            });
+            ctx->prober.on_error([&ctx](Error) { ctx->reactor->break_loop(); });
+            ctx->prober.send_probe(mk::jni::cxxstring(env, destIp), destPort,
+                                   ttl, payload, timeout);
         });
-        ctx->prober.on_timeout([&ctx, &is_timed_out]() {
-            ctx->poller.break_loop();
-            is_timed_out = true;
-        });
-        ctx->prober.on_error([&ctx](Error) { ctx->poller.break_loop(); });
-        ctx->prober.send_probe(mk::jni::cxxstring(env, destIp), destPort, ttl,
-                               payload, timeout);
-        ctx->poller.loop();
```

Comment:
This hunk is significantly simpler if we read [the diff without considering spaces](https://github.com/measurement-kit/measurement-kit-android/pull/9/files?w=1).